command/debug.go

```diff
@@ -551,28 +615,51 @@ func (c *DebugCommand) collectConsul(dir, consul string) error {
 
 // collectVault calls the Vault API directly to collect data
 func (c *DebugCommand) collectVault(dir, vault string) error {
-	if vault == "" {
+	addr := c.vault.addr(vault)
+	if addr == "" {
 		return nil
 	}
 
-	token := c.vaultToken
-	if token == "" {
-		os.Getenv("VAULT_TOKEN")
-	}
-
-	client := http.Client{
-		Timeout: 2 * time.Second,
-	}
+	client := api.DefaultHttpClient()
+	api.ConfigureTLS(client, c.vault.tls)
 
-	req, _ := http.NewRequest("GET", vault+"/sys/health", nil)
-	req.Header.Add("X-Vault-Token", token)
+	req, _ := http.NewRequest("GET", addr+"/sys/health", nil)
+	req.Header.Add("X-Vault-Token", c.vault.token())
 	req.Header.Add("User-Agent", userAgent)
 	resp, err := client.Do(req)
 	c.writeBody(dir, "vault-sys-health.json", resp, err)
 
 	return nil
 }
 
+func (c *external) addr(addr string) string {
```

Comment:
I'd suggest colocating `external` methods and structs together; maybe at the bottom of the file?  Also, `e` might be a better abbreviation for `external`.

Reply:
done!